nova/conductor/manager.py

```diff
@@ -312,6 +312,14 @@ def _get_request_spec_for_cold_migrate(context, instance, flavor,
             sh = filter_properties.setdefault('scheduler_hints', {})
             sh.update(new_scheduler_hints)
 
+        volume_sizes = [bdm.volume_size for bdm in instance.get_bdms()
+                        if bdm.destination_type == 'volume']
+        sh = filter_properties.setdefault('scheduler_hints', {})
+        # NOTE(jkulik): We have to update the "volumes" key even if there are
+        # no volumes attached currently, because the "scheduler_hints" are
+        # persisted in the DB and thus could contain old information.
+        sh.update({'volumes': volume_sizes})
```

Comment:
I am not too sure about the name of the field. If it is  `volume_sizes` , then I would expect the field to be namd the same.

Wouldn't it make sense to store there right away more information?  We discussed using the volume shard to prefer the same shard.

While this PR is limited to resize, and we talked about the instance creation, I would prefer to have the same data and same fields.

Reply:
I looked for a ticket, because I remembered the same, but couldn't find one. Therefore, I opted for solving that specific ticket.

BDMs, afaict, contain the `connection_info` only after the scheduling, so we would have to talk to Cinder to get information about the volumes' shards during VM creation. I'd put that into `nova.compute.api` instead of the conductor. I do not think the shard information helps in resizing though.

I agree, that we should rename the variable to `volume_sizes` as I learned later on that `scheduler_hints` is a `DictOfListOfStringsField` and thus we can't put any more complex information into them - unless we store them as JSON. I'd be open to find another way of transporting the sizes, too, but haven't found anything sensible, yet. Given that, I'd think we'd might use a new key `volume_shards` - even though a good scheduling decision should take into account sizes and shards at the same time.

So yeah, not sure how to continue. I don't really want to add additional calls to Cinder unless we work on a filter to consume that information, because it creates unnecessary overhead.

Reply:
Then, I would say we stay with `volume_sizes`, and revisit as soon as we know better how to solve the initial scheduling problem. With some luck, we will be on NFS then anyway.